src/SoundOptionsModal.js

```diff
@@ -0,0 +1,138 @@
+};
+
+type SoundOptionsModalStates = {
+    audioEnabled: boolean,
+    announcementsEnabled: boolean,
+    sonificationEnabled: boolean
+};
+
+class SoundOptionsModal extends React.Component<SoundOptionsModalProps, SoundOptionsModalStates> {
+    constructor (props: SoundOptionsModalProps) {
+        super(props);
+        this.state = {
+            audioEnabled: this.props.audioEnabled,
+            announcementsEnabled: this.props.announcementsEnabled,
+            sonificationEnabled: this.props.sonificationEnabled
+        };
+    };
+
+    handleToggleAudioEnabled = () => {
+        this.setState({
+            audioEnabled: !this.state.audioEnabled
+        });
+    };
+
+    handleToggleAnnouncementsEnabled = () => {
+        this.setState({
+            announcementsEnabled: !this.state.announcementsEnabled
+        });
+    };
+
+    handleToggleSonificationEnabled = () => {
+        this.setState({
+            sonificationEnabled: !this.state.sonificationEnabled
+        });
+    };
+
+    handleOnSave = () => {
+        this.props.onChangeSoundOptions(this.state.audioEnabled, this.state.announcementsEnabled, this.state.sonificationEnabled );
+    }
+
+    handleOnCancel = () => {
+        this.setState({
+            audioEnabled: this.props.audioEnabled,
+            announcementsEnabled: this.props.announcementsEnabled,
+            sonificationEnabled: this.props.sonificationEnabled
+        });
+        this.props.onCancel();
+    }
+
+    renderSoundOptions = () => {
+        const soundOptions = [];
+        const soundOptionsProps = [
+            { id:'sound-options-allsounds', label: this.props.intl.formatMessage({ id: 'SoundOptionsModal.allSounds'}), toggleState: this.state.audioEnabled, onChange: this.handleToggleAudioEnabled },
+            { id:'sound-options-musicalSounds', label: this.props.intl.formatMessage({ id: 'SoundOptionsModal.musicalSounds' }), toggleState: this.state.sonificationEnabled, onChange: this.handleToggleSonificationEnabled },
+            { id:'sound-options-announcements', label: this.props.intl.formatMessage({ id: 'SoundOptionsModal.announcements' }), toggleState: this.state.announcementsEnabled, onChange: this.handleToggleAnnouncementsEnabled }
+        ];
+        for (let i=0; i< soundOptionsProps.length; i++) {
+            const soundOptionProps = soundOptionsProps[i];
+            soundOptions.push(
+                <div className='SoundOptions__option-container' key={soundOptionProps.id}>
+                    <div aria-hidden={true}>
+                        {soundOptionProps.label}
+                    </div>
+                    <div className='SoundOptions__toggle-container'>
+                        <div aria-hidden={true}>
+                            {this.props.intl.formatMessage({ id: 'SoundOptionsModal.toggleOff' })}
+                        </div>
+                        <ToggleSwitch
```

Comment:
I was trying to understand this in relation to other recent modals.  Some of them (like actions simplification) prevent the default on events they respond to.  The `ToggleSwitch` used here prevents default for keyboard events, but not clicks.  We should talk about why we prevent default and lightly standardise some of the common components like `ToggleSwitch` and `IconButton` as needed.